compiler/rustc_resolve/src/late.rs

```diff
@@ -1454,72 +1594,76 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
                     //
                     //     impl Foo for std::cell::Ref<u32> // note lack of '_
                     //     async fn foo(_: std::cell::Ref<u32>) { ... }
-                    LifetimeRibKind::AnonymousCreateParameter(_) => {
+                    LifetimeRibKind::AnonymousCreateParameter { report_in_path: true, .. }
+                        // Do not report an error if we wouldn't even lint.
+                        if should_lint =>
+                    {
+                        let sess = self.r.session;
+                        let mut err = rustc_errors::struct_span_err!(
+                            sess,
+                            path_span,
+                            E0726,
+                            "implicit elided lifetime not allowed here"
+                        );
+                        rustc_errors::add_elided_lifetime_in_path_suggestion(
+                            sess.source_map(),
+                            &mut err,
+                            expected_lifetimes,
+                            path_span,
+                            !segment.has_generic_args,
+                            elided_lifetime_span,
+                        );
+                        err.note("assuming a `'static` lifetime...");
+                        err.emit();
+                        should_lint = false;
+
+                        for i in 0..expected_lifetimes {
+                            let id = node_ids.start.plus(i);
+                            self.record_lifetime_res(id, LifetimeRes::Error);
+                        }
+                        break;
+                    }
+                    // Do not create a parameter for patterns and expressions.
+                    LifetimeRibKind::AnonymousCreateParameter { binder, .. } if should_lint => {
+                        let res = self.create_fresh_lifetime(node_ids.start, ident, binder);
+                        self.record_lifetime_res(node_ids.start, res);
+                        for i in 1..expected_lifetimes {
```

Comment:
```suggestion
                        for id in node_ids {
```
Also not clear why the first node id is treated separately here.